### PR TITLE
another readme rewrite

### PR DESCRIPTION
**Pull request by rmorshea**
Yet another take on the README. Not sure I like it yet, but I think it's more inline with how I imagine it looking in the end.

### CODE REVIEW

File: src/idom/_option.py [hunk 1]
```diff
@@ -107,6 +107,7 @@ def _current(self) -> _O:  # type: ignore
         warnings.warn(
             f"{self.name!r} has been renamed to {self._new_opt.name!r}",
             DeprecationWarning,
+            stacklevel=3,
```
Comment (rmorshea) [inline src/idom/_option.py:110]: Not related, just something I and wanted to improve caught.

**Review by Archmonger — CHANGES_REQUESTED** [review]
I personally dislike this one. It really doesn't give a top-level highlight of the main reasons why anyone would ever want to use IDOM.

The main thing about the readme is trying to "sell" IDOM to people who haven't read the docs yet. We need to highlight all our selling points as concisely as possible. 90% of people will never read the docs.

In my previous PR, I listed the main things that I consider selling points for new developers.

Also, we definitely want to avoid tech jargon in any readme we create. Words such as "declarative" shouldn't make it in.

**Comment by rmorshea** [comment]
Good point.

**Comment by rmorshea** [comment]
I think the README more directly gets at the features IDOM provides, but it's also more scattershot now. Maybe there's too much that we're trying to fit into it?

Rather than throwing all the features at the reader at once we focus on something like 3:

- Write apps without JS
- Integration with web frameworks like FastAPI, Django, etc
- React-style components in Python

@acivitillo or @liberty-rising do you have some thoughts on what the reader our should come away with from the README?

**Comment by rmorshea** [comment]
I'm gonna close this and open a discussion.

**Pull request closed without merging**